Source/ClientObserver.swift

```diff
@@ -6,11 +6,11 @@
 import Foundation
 
 class ClientObserver {
-	typealias LightsDidUpdate = (lights: [Light]) -> Void
-
-	let lightsDidUpdateHandler: LightsDidUpdate
-
-	init(lightsDidUpdateHandler: LightsDidUpdate) {
+	typealias LightsDidUpdate = (_ lights: [Light]) -> Void
+	
+	let lightsDidUpdateHandler: LightsDidUpdate?
```

Comment:
What's the motivation for making this property optional? The original intent behind this class was to let you bind observers to listen for changes to the client. If you intend to have a nil callback, why not just skip setting up an observer?

Would you be able to share a use case?